plugins/woocommerce/includes/gateways/paypal/includes/class-wc-gateway-paypal-request.php

```diff
@@ -549,7 +555,8 @@ private function get_paypal_shipping_preference( $order ) {
 	 * Get the shipping information for the PayPal create-order request.
 	 *
 	 * @param WC_Order $order Order object.
-	 * @return array
+	 * @return array|null Returns null if the shipping is not required,
+	 *  or the address is not set, or is incomplete.
 	 */
 	private function get_paypal_order_shipping( $order ) {
 		if ( ! $order->needs_shipping_address() ) {
```

Comment:
```suggestion
		if ( ! $order->needs_shipping() ) {
```

Let's change it to `needs_shipping`. I mistakenly added `needs_shipping_address` here in one of my PRs. But the logic should depend on whether the cart items need shipping or not which is covered in `needs_shipping` method.

I came across a scenario where we are sending `No Shipping` to PayPal wrongly and PayPal is not showing the shipping options for my selected physical product.

<img width="633" height="736" alt="Screenshot 2025-09-23 at 4 10 35 PM" src="https://github.com/user-attachments/assets/9b7f78a8-cbb8-46cb-bbe8-9c595319e69f" />

Checking for `needs_shipping` here instead, fixes the issue.

Reply:
We need to replace it in `get_paypal_shipping_preference` method as well.